bpo-28963: Fix out of bound iteration in asyncio.Future.remove_done_callback
implemented in C.